### PR TITLE
fix: bc history notes display across viewports

### LINKED ISSUE #2692: Dropdown content box in bc tables not filling entire table width
While I was working on some alignment with the star icons in bc tables, I noticed the dropdown content box doesn't occupy the entire width of the table which seemed a bit odd. This might not be an issue, but I wanted to bring it up just in case. 

It's reproduced for other pages but in the bc table on [:nth-child() - CSS](https://developer.mozilla.org/en-US/docs/Web/CSS/:nth-child) is where the screenshot from below comes from.

Screen Shot 2021-01-25 at 10 25 42 PM

From what I can see, this is intended for the text to have `max-width: 75ch`

Screen Shot 2021-01-25 at 10 28 09 PM

What are your thoughts on keeping `max-width` for the text but have the background color still occupy 100% width of the table cell like the "Matches element with no parent" box does shown in the picture.

### LINKED ISSUE #3076: BCD note  layout is broken in visibilitychange event page
Visit https://developer.mozilla.org/en-US/docs/Web/API/Document/visibilitychange_event#browser_compatibility and click the disclosure arrow for Safari, which is showing partial support:

Screen Shot 2021-02-26 at 2 00 48 PM

### PR DESCRIPTION
**Pull request by schalkneethling**
The fixes the items highlighted in #2692 and #3076. It also includes some minor updates to how the icon is displayed which exposes the bc history notes.

Fix #2692
Fix #3076

### CODE REVIEW

**Comment by tannerdolby** [comment]
@schalkneethling I pulled into this feature branch and tested a few pages locally. It looks like #3076 is fixed for the most part, it looks really nice above below 1199px and above 1340px. But still there is a problematic region between 1199px and 1340px [1199, 1340) where the first `<dd>` element is super squished. Other than that, I'd reckon 3076 is fixed.

http://localhost:3000/en-US/docs/Web/API/Document/visibilitychange_event

Screen Shot 2021-04-29 at 12 40 16 AM

**Comment by tannerdolby** [comment]
In regards to #2692, after testing locally, in between 802px and 1024px the extra space still exists to the right of `.bc-notes-wrapper` due to the `max-width: 85ch` on `.main-page-content dl` from `index.scss`. I'm going to continue looking through your PR and try to find anywhere where we could fix this.

Screen Shot 2021-04-29 at 12 58 07 AM

**Comment by schalkneethling** [comment]
> But still there is a problematic region between 1199px and 1340px [1199, 1340) where the first <dd> element is super squished. 

@tannerdolby thanks for testing this! This is odd but perhaps I can just set the `flex` for those to not shrink and only grow. Alternatively, we can just keep the stacked layout until we reach "large desktop". Will have a look what looks/works best.

**Comment by schalkneethling** [comment]
> In regards to #2692, after testing locally, in between 802px and 1024px the extra space still exists to the right of .bc-notes-wrapper

Dang it! 🙃  There clearly is a little in between where the override does not kick in.

**Comment by schalkneethling** [comment]
Wait, wait. This PR is busted at the moment. Fixing.

**Comment by tannerdolby** [comment]
> @tannerdolby thanks for testing this!

@schalkneethling your welcome! It looks like you fixed up the PR and all checks are passing, I will sync my local copy and then test a few pages out.

**Comment by schalkneethling** [comment]
> > @tannerdolby thanks for testing this!
> 
> @schalkneethling your welcome! It looks like you fixed up the PR and all checks are passing, I will sync my local copy and then test a few pages out.

No 😀 don't test yet. I mixed up two pull requests. I have not made updates. Will let you know as soon as it is ready for a second look. Thanks, @tannerdolby

**Comment by tannerdolby** [comment]
@schalkneethling Ok that's what I thought 😄, I was positive I hadn't seen any commits pushed on this branch but a commit being forced pushed to another branch. 

> Will let you know as soon as it is ready for a second look.

Sounds good, chat again soon!

**Comment by schalkneethling** [comment]
> @schalkneethling Ok that's what I thought 😄, I was positive I hadn't seen any commits pushed on this branch but a commit being forced pushed to another branch.
> 
> > Will let you know as soon as it is ready for a second look.
> 
> Sounds good, chat again soon!

Ready for another look @tannerdolby

**Review by tannerdolby — APPROVED** [review]
Looks great! Both #2692 and #3076 are resolved 🎉

**Comment by schalkneethling** [comment]
> Really nice work @schalkneethling!
> 
> Both #2692 and #3076 are resolved 🎉

Yippeee! @Gregoor is going to also take it for a spin in a bit and then we can button it up as they say ;p - Thanks @tannerdolby

**Comment by tannerdolby** [comment]
Woohooo! I'm excited we got this all sorted. Your welcome @schalkneethling

File: client/src/document/ingredients/browser-compatibility-table/atoms/_bc-history-link.scss [hunk 1]
```diff
@@ -23,7 +23,10 @@
       content: "";
       display: block;
       height: 16px;
+      left: 50%;
       margin: 0 auto;
+      position: absolute;
```
Comment (Gregoor) [inline client/src/document/ingredients/browser-compatibility-table/atoms/_bc-history-link.scss:28]: I am probably blowing up the scope of this issue, but looking at the BCD's CSS I started to wonder why we need this pseudo-element here (and a lot of its styles here).
I started playing with deleting it and just putting a centered background-image on the `.bc-history-link`. I then also had to `transform: rotate(180deg);` the element itself and flip the border position but it looked like basically the same thing with a whole lot less CSS.

I probably missed edge cases there and it is not quite what this issue is about, so do with that what you will.

**Review by Gregoor — APPROVED** [review]
Looks good, just had some thoughts while looking at the CSS. Do with them what you want!

File: client/src/document/ingredients/browser-compatibility-table/atoms/_bc-history-link.scss (hunk 1 above)
Comment (schalkneethling) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-history-link.scss:28]: Hmmm. Yeah, I had not really interrogated it. Let me have a look at it tomorrow and perhaps we can indeed clean up some.

Comment (schalkneethling) [reply client/src/document/ingredients/browser-compatibility-table/atoms/_bc-history-link.scss]: Addressed this. I bet there is a lot of things like this we can do to improve the BCD table CSS. I have never really touched it in the transition phase to Yari. It is such a tricky component sometimes but, I would love to dig into it and clean up/improve for sure. Thanks for bringing this to my attention Gregor.

**Pull request merged**